### PR TITLE
Create contest #47

### PR DESCRIPTION
**Pull request by yliu298**
### What this PR does (required):
- Create new contest in create-new-contest page

### Screenshots / Videos (required):
- 
Screenshot 2021-07-22 184131

### Any information needed to test this feature (required):
click create contest button on navBar
input require input fields
click create button

### Any issues with the current functionality (optional):
when validate time and timezone using Yup, the onSubmit function will not work
DatePicker is not returned as selected date
date, time, and timezone can not convert to as deadline

### CODE REVIEW

File: client/src/components/DesignSubmit/DesignSubmit.tsx [hunk 1]
```diff
@@ -66,7 +66,9 @@ const DesignSubmit = ({ contest }: Props): JSX.Element => {
       updateSnackBarMessage('Your design images have been uploaded successfully');
     }
     if (result.success) {
-      const returnData = await contestImgSubmitAPI(contest._id, result.success.urlArray);
+      //for testing by adding contest id, will change back to parmas.id when contest details page done
+      const returnData = await contestImgSubmitAPI('60fa08e84f14460a342ad347', result.success.urlArray);
+      console.log(returnData);
```
Comment (rajivtitus) [inline client/src/components/DesignSubmit/DesignSubmit.tsx:71]: Please remove

File: client/src/helpers/APICalls/contest.ts [hunk 2]
```diff
@@ -1,5 +1,26 @@
+import { AuthApiData } from '../../interface/AuthApiData';
 import { FetchOptions } from '../../interface/FetchOptions';
 
+export async function createContestAPI(
+  title: string,
+  description: string,
+  prizeAmount: string,
+  deadline: Date | null,
```
Comment (rajivtitus) [inline client/src/helpers/APICalls/contest.ts]: Why are we using type null here?

File: client/src/pages/Contest/NewContest.tsx [hunk 3]
```diff
@@ -32,11 +35,23 @@ const NewContest: FunctionComponent = (): JSX.Element => {
...
       prizeAmount: string;
-      date: string;
+      deadline: Date | null;
       time: string;
       timezone: string;
     }>,
   ) => {
+    createContestAPI(title, description, prizeAmount, deadline).then((data) => {
+      if (data.error) {
+        setSubmitting(false);
```
Comment (rajivtitus) [inline client/src/pages/Contest/NewContest.tsx]: I don't think we to `setSubmitting` to false here and below, as it's being set outside the else/if construct (line 55)?

**Review by rajivtitus — APPROVED** [review]
LGTM! Just a couple of comments, please address before merging

File: client/src/components/DesignSubmit/DesignSubmit.tsx (hunk 1 above)
Comment (yliu298) [reply client/src/components/DesignSubmit/DesignSubmit.tsx:71]: removed

File: client/src/helpers/APICalls/contest.ts (hunk 2 above)
Comment (yliu298) [reply client/src/helpers/APICalls/contest.ts]: remove null in API calls and component definition

File: client/src/pages/Contest/NewContest.tsx (hunk 3 above)
Comment (yliu298) [reply client/src/pages/Contest/NewContest.tsx]: removed setSubmitting(false) inside if/else condition.

**Pull request merged**